api/api.go

```diff
@@ -0,0 +1,108 @@
+package api
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/NebulousLabs/Sia/types"
+	"github.com/NebulousLabs/Sia/modules"
+)
+
+// As this modules interacts heavily with the Sia API, which sends
+// structs over json, this package uses the simply imports the module
+// packgae where those structs are defined in the core sia package
+
+type ApiLink struct {
+	url string
+
+	// Port is a string as both http requests and command line arguments are strings
+	port string
+}
```

Comment:
also I believe this type already exists in the network package, a 'NetAddress' or something like that.